src/main/java/com/SAS/soccer_association_system/UserAPIController.java

```diff
@@ -21,25 +22,36 @@
     /**
      * The function receives username and password and returns response OK if the exists in the system,
      * otherwise returns false
+     *
      * @return the list
      */
-    @PostMapping(value ="/login")
+    @PostMapping(value = "/login")
     public String postUser(@RequestBody String credentials, HttpServletRequest request) {
         JSONObject json = new JSONObject(credentials);
         String username = json.get("username").toString();
         String password = json.get("password").toString();
         String clientURL = request.getRemoteAddr() + ":" + request.getRemotePort();
         String role = app.login(username, password, clientURL);
-        return role.isEmpty() ? "Failed" : role;
+        if (!role.isEmpty()) {
+            JSONObject toReturn = new JSONObject();
+            toReturn.put("role", role);
+            String userId = UsersCRUD.getUserIdByUserName(username) + "";
```

Comment:
Same here, you can just change the login function - app.login, that will return List of string or maybe a pair.

Reply:
working with JSON is easyer